sunpy/map/mapbase.py

```diff
@@ -225,7 +225,10 @@ def __init__(self, data, header, plot_settings=None, **kwargs):
         params = list(inspect.signature(NDData).parameters)
         nddata_kwargs = {x: kwargs.pop(x) for x in params & kwargs.keys()}
         super().__init__(data, meta=MetaDict(header), **nddata_kwargs)
-
+        # Check for missing 'rsun_ref' in header and add it if necessary
+        if 'rsun_ref' not in self.meta:
+            # Default value for solar radius in kilometers (696340 km)
+            self.meta['rsun_ref'] = 696340 * u.km.to(u.m)
```

Comment:
For me, this is the wrong approach, the radius of the sun is not constant. So we can't set it by default at the init of a map.

Reply:
Can we implement a check to use the manually provided solar radius if available, and fall back to the default value if not provided?

Reply:
Within mapbase?